src/js/node/fs.ts

```diff
@@ -1021,6 +1021,10 @@ function _toUnixTimestamp(time: any, name = "time") {
 function opendirSync(path, options) {
   // TODO: validatePath
   // validateString(path, "path");
+  if (!fs.existsSync(path)) {
```

Comment:
Let's use `fs.accessSync`

![CleanShot 2025-03-03 at 15 09 55@2x](https://github.com/user-attachments/assets/c0693f26-5923-4cff-875d-274d0bb0bb72)

Reply:
accessSync won't error if it's not a directory

Reply:
true. this should expose an opendir binding in `src/bun.js/node/node_fs_binding.zig`, a matching implementation function in `src/bun.js/node/node_fs.zig`. this function would return a directory file descriptor. Then, there can be internal variants of `readdir` that work on this file descriptor.

Reply:
Thanks for the pointers. I figured this would involve some Zig bindings, but wanted to ensure that there wasn't anything in JS-land that I was missing. I'm starting to work on the bindings now, should hopefully have something up this week, time permitting